src/phase0/state_transition/mod.rs

```diff
@@ -631,3 +631,104 @@ pub fn get_active_validator_indices<
     }
     active
 }
+
+pub fn get_validator_churn_limit<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+    const CHURN_LIMIT_QUOTIENT: usize,
+    const MIN_PER_EPOCH_CHURN_LIMIT: usize,
+>(
+    state: &BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    context: &Context,
+) -> u64 {
+    let active_validator_indices =
+        get_active_validator_indices(state, get_current_epoch(state, context));
+    u64::max(
+        MIN_PER_EPOCH_CHURN_LIMIT as u64,
+        active_validator_indices.len() as u64 / CHURN_LIMIT_QUOTIENT as u64,
+    )
+}
+
+pub fn get_seed<
+    const SLOTS_PER_HISTORICAL_ROOT: usize,
+    const HISTORICAL_ROOTS_LIMIT: usize,
+    const ETH1_DATA_VOTES_BOUND: usize,
+    const VALIDATOR_REGISTRY_LIMIT: usize,
+    const EPOCHS_PER_HISTORICAL_VECTOR: usize,
+    const EPOCHS_PER_SLASHINGS_VECTOR: usize,
+    const MAX_VALIDATORS_PER_COMMITTEE: usize,
+    const PENDING_ATTESTATIONS_BOUND: usize,
+    const MIN_SEED_LOOKAHEAD: u64,
+>(
+    state: &BeaconState<
+        SLOTS_PER_HISTORICAL_ROOT,
+        HISTORICAL_ROOTS_LIMIT,
+        ETH1_DATA_VOTES_BOUND,
+        VALIDATOR_REGISTRY_LIMIT,
+        EPOCHS_PER_HISTORICAL_VECTOR,
+        EPOCHS_PER_SLASHINGS_VECTOR,
+        MAX_VALIDATORS_PER_COMMITTEE,
+        PENDING_ATTESTATIONS_BOUND,
+    >,
+    epoch: Epoch,
+    domain_type: DomainType,
+) -> Bytes32 {
+    let epoch = epoch + (EPOCHS_PER_HISTORICAL_VECTOR as u64 - MIN_SEED_LOOKAHEAD) - 1;
+    let mix = get_randao_mix(state, epoch);
+    let mut input = [0u8; 44];
+    input[..4].copy_from_slice(&domain_type.as_bytes());
+    input[4..].copy_from_slice(&epoch.to_le_bytes());
```

Comment:
i think `copy_from_slice` will panic if the `src` and `dst` are not the same length so you want to properly size the `input[4..8]` here

Reply:
fixed